### PR TITLE
Feature: Build Docker images with all necessary environment stuff

### PR DESCRIPTION
**Pull request by prokopsimek**
## Description

Added CI stage to build Docker image with all necessary environment to run the DX Scanner.

## Motivation and Context

The user now doesn't have to download all prerequisites and can just use prepared docker image.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly.
- [ ] ~I have added new practice to practice list in README.md.~
- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [ ] ~I have added tests to cover my changes.~
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by prokopsimek** [comment]
# [DX Scanner](https://dxscanner.io) Report <a href="https://dxscanner.io" target="_blank">DXScanner.io</a>
## https://github.com/DXHeroes/dx-scanner 

<h3>⚠️ Improvements with medium impact
</h3>
<details>
<summary>Break down large pull requests into smaller ones</summary>

<b>Large pull request are hard to code review and it reduces the probability of finding bugs. Split your PRs into logical units. Do not have PR with more than 1000 changes.</b> <a href="https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067">https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067</a>
</details>
<details>
<summary>Solve Issues Continuously</summary>

<b>Do not have an open Issues more than 60 days. Solve Issues continuously.</b> <a href="https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612">https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612</a>
</details>
<details>
<summary>Update Dependencies of Minor and Patch Level</summary>

<b>Keep the dependencies updated to eliminate security concerns and compatibility issues. Use, for example, npm-check-updates.</b> <a href="https://dxkb.io/p/updating-the-dependencies">https://dxkb.io/p/updating-the-dependencies</a>


 Name                             | New    | Current  
---------------------------------- | -------- | ----------
 cross-env                        | 7.0.2  | 7.0.0    
 npm-check-updates                | 4.0.4  | 4.0.1    
 tslib                            | 1.11.1 | 1.11.0   
 typescript                       | 3.8.3  | 3.8.2    
 @types/node                      | 13.7.7 | 12.12.28 
 nock                             | 12.0.2 | 12.0.1   
 @types/node                      | 13.7.7 | 12.12.28 
 @typescript-eslint/eslint-plugin | 2.22.0 | 2.21.0   
 @typescript-eslint/parser        | 2.22.0 | 2.21.0   

  <b>Impact changed from high to medium.</b>
</details>
<h3>🔔 Improvements with minor impact
</h3>
<details>
<summary>Write Commit Messages by Convention</summary>

<b></b> <a href="https://www.conventionalcommits.org/">https://www.conventionalcommits.org/</a>


 Commit Message                                                                | Problems                                                                                                                                   
------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------
 fix(practice): SecurityVulnerabilitiesPractice won't crash on npm audit error | header must not be longer than 72 characters, current length is 77; subject must not be sentence-case, start-case, pascal-case, upper-case 
                                                                               |                                                                                                                                            
 This could happen e.g. if you have outdated lockfile.                         |                                                                                                                                            
 fix: relocated interfaces of individual practices in the current architecture | header must not be longer than 72 characters, current length is 77                                                                         

</details>
<details>
<summary>Update Dependencies of Major Level</summary>

<b>Keep the dependencies updated to have all possible features. Use, for example, npm-check-updates.</b> <a href="https://dxkb.io/p/updating-the-dependencies">https://dxkb.io/p/updating-the-dependencies</a>


 Name          | New     | Current  
--------------- | --------- | ----------
 @octokit/rest | ^17.0.0 | ^16.0.0  
 @types/node   | 13.7.7  | 12.12.28 

</details>
<details>
<summary>Use a JS Logging Library</summary>

<b>Use a logging library to avoid errors and even cyber attacks. The most widely used logging library in the JavaScript community is Winston.</b> <a href="https://www.npmjs.com/package/winston/">https://www.npmjs.com/package/winston/</a>
</details>



---



Implementation is not adoption.
We can help you with both. :-)
[dxheroes.io](https://dxheroes.io)



##### Found a bug? Please <a href="https://github.com/DXHeroes/dx-scanner/issues" target="_blank">report</a>.

**Comment by prokopsimek** [comment]
- [ ] the image run doesn't work as is desribed in CONTRIBUTION.md (`docker run -it -v `pwd`:/usr/app dxs dx-scanner run .`)

File: entrypoint.sh [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+if [[ $1 == *"bash" ]]; then
+  exec "/bin/bash"
+  exit 0;
+elif [[ $1 == *"sh" ]]; then
+  exec "/bin/sh"
+  exit 0;
+fi
 
-dx-scanner run --ci -r -a $1
+echo "Start DX Scanner Action"
```
Comment (adelkahomolova) [inline entrypoint.sh]: It doesn't have to be always run as a action

File: entrypoint.sh [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+if [[ $1 == "dx-scanner "* || $1 == "dxs "* || $1 == "dxscanner "* ]]; then
+  echo $@
+  exec "$@"
+elif [[ $1 == "-"* ]]; then
+  echo dx-scanner run $@
+  exec dx-scanner run "$@"
```
Comment (adelkahomolova) [inline entrypoint.sh]: It's not possible to get help just for dx-scanner if I want to pass just `--help`. It runs help for dx-scanner run

File: entrypoint.sh [hunk 3]
```diff
@@ -1,5 +1,22 @@
-#!/bin/sh -l
+#!/bin/bash
+set -e
 
-echo "Start DX Scanner Action"
+if [[ $1 == *"bash" ]]; then
+  exec "/bin/bash"
+  exit 0;
```
Comment (adelkahomolova) [inline entrypoint.sh]:
```suggestion
```

It's not necessary because of `exec`

File: entrypoint.sh [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+elif [[ $1 == *"sh" ]]; then
+  exec "/bin/sh"
+  exit 0;
```
Comment (adelkahomolova) [inline entrypoint.sh]:
```suggestion
```

It's not necessary because of exec

File: entrypoint.sh [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+fi
 
-dx-scanner run --ci -r -a $1
+echo "Start DX Scanner Action"
+if [[ $1 == "dx-scanner "* || $1 == "dxs "* || $1 == "dxscanner "* ]]; then
+  echo $@
+  exec "$@"
```
Comment (adelkahomolova) [inline entrypoint.sh]: It doesn't work. It's impossible to pass url in any option.

File: entrypoint.sh [hunk 6]
```diff
@@ -1,5 +1,22 @@
...
+elif [[ $1 == "-"* ]]; then
+  echo dx-scanner run $@
+  exec dx-scanner run "$@"
+else
+  echo dx-scanner run --ci -r -a $@
+  exec dx-scanner run --ci -r -a "$@"
```
Comment (adelkahomolova) [inline entrypoint.sh]: It's impossible to run it for remote url

**Comment by adelkahomolova** [comment]
Also: 

- [ ] running without parameters doesn't work

- [ ] docker run --rm -it dxs run https://github.com/DXHeroes/empty-js doesn't work

- [ ] docker run --rm -it dxs dx-scanner run test doesn't work

**Comment by prokopsimek** [comment]
> Also:
> 
> * [ ]  running without parameters doesn't work
> * [ ]  docker run --rm -it dxs run https://github.com/DXHeroes/empty-js doesn't work
> * [ ]  docker run --rm -it dxs dx-scanner run test doesn't work

I have fixed all in-code comments.

I have a note to your list:
- `running without parameters doesn't work`
It works for now but we have a bug in `--ci` mode https://github.com/DXHeroes/dx-scanner/issues/260

- `docker run --rm -it dxs run https://github.com/DXHeroes/empty-js doesn't work`
it's invalid command, the first `dxs` must be name of a tagged container and there is missing second `dxs` command. `docker run --rm -it dxs dxs run https://github.com/DXHeroes/empty-js` is correct command with your case and it works for now.

- `docker run --rm -it dxs dx-scanner run test`
Works for now. But we should make better error handling.

**Comment by prokopsimek** [comment]
@adelkahomolova And it's necessary to use `-v `pwd`:/usr/app` as an argument to pass your local code to the container as a volume.

**Pull request merged**